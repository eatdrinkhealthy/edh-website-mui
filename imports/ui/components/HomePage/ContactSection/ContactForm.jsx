import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";

const styles = {
  form: {
    border: "1px dashed green",
    textAlign: "center",
  },
};

const ContactForm = props => {
  const {
    classes: { form },
  } = props;

  return <div className={form}>Contact Form</div>;
};

ContactForm.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default withStyles(styles)(ContactForm);
