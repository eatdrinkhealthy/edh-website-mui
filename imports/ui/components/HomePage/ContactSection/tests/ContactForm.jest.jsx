// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { mount } from "enzyme";
import { createStore } from "redux";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import ContactForm from "../ContactForm";
import mountFormWithInputs from "../../../../../../tests/mountFormWithInputs";
import appReducer from "../../../../../state/reducers";

describe("<ContactForm />", () => {
  const testStore = createStore(appReducer);

  it("matches render snapshot - with no form validation errors", () => {
    // render form with no input values to display validation errors
    const tree = renderer
      .create(
        <Provider store={testStore}>
          <ContactForm onSubmit={() => {}} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it.skip("NEW: should set Name 'too short' error, when Name less than min length", () => {
    const testStore2 = createStore(appReducer);
    const props = {
      onSubmit: jest.fn(),
    };

    const wrapper = mount(
      <Provider store={testStore2}>
        <ContactForm {...props} />
      </Provider>,
    );

    const name = wrapper.find("TextField[name='contactName']");
    name.simulate("change", { target: { value: "a" } });
    name.simulate("blur");
    console.log("input value:", name.props());

    console.log(name.debug());
  });

  it.skip("should set Name 'too short' error, when Name less than min length", () => {
    const props = {
      onSubmit: jest.fn(),
    };

    const wrapper = mountFormWithInputs(
      <ContactForm {...props} />,
      {
        contactName: "a",
      },
      testStore,
    );

    wrapper.find("input[name='contactName']").simulate("blur"); // blur triggers field validation

    expect(wrapper.find("FormHelperText").text()).toBe("Must have a least 2 characters.");
  });

  it.skip("should set invalid email error, when email format is invalid", () => {
    const props = {
      onSubmit: jest.fn(),
    };

    const wrapper = mountFormWithInputs(
      <ContactForm {...props} />,
      {
        contactEmail: "user12",
      },
      testStore,
    );

    wrapper.find("input[name='contactEmail']").simulate("blur"); // blur triggers field validation

    const helperTexts = wrapper.find("FormHelperText");
    console.log("helper texts:", helperTexts.debug());

    expect(wrapper.find("FormHelperText").text()).toBe("Invalid email address.");
  });

  it.skip("should set message 'too short' error, when message is too short", () => {
    const props = {
      onSubmit: jest.fn(),
    };

    const wrapper = mountFormWithInputs(
      <ContactForm {...props} />,
      {
        contactMessage: "1",
      },
      testStore,
    );

    wrapper.find("#contactMessage").simulate("blur"); // blur triggers field validation

    expect(wrapper.find("#contactMessageError").text()).toBe(
      "Must have a least 2 characters.",
    );
  });

  it.skip("should set message 'too long' error, when message is too long", () => {
    const props = {
      onSubmit: jest.fn(),
    };

    const wrapper = mountFormWithInputs(
      <ContactForm {...props} />,
      {
        contactMessage:
          "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890" +
          "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890" +
          "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890" +
          "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890" +
          "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890",
      },
      testStore,
    );

    wrapper.find("#contactMessage").simulate("blur"); // blur triggers field validation

    expect(wrapper.find("#contactMessageError").text()).toBe(
      "Must not have more than 500 characters.",
    );
  });

  it.skip("should call onSubmit, when submitting and all form fields are valid", () => {
    const props = {
      onSubmit: jest.fn(),
    };

    const wrapper = mountFormWithInputs(
      <ContactForm {...props} />,
      {
        contactName: "user12",
        contactEmail: "user12@test.com",
        contactMessage: "a message to send.",
      },
      testStore,
    );

    wrapper.find("#contactSubmit").simulate("submit");
    expect(props.onSubmit).toHaveBeenCalledWith(
      {
        contactName: "user12",
        contactEmail: "user12@test.com",
        contactMessage: "a message to send.",
      },
      expect.anything(),
      expect.anything(),
    );
  });

  it.skip("should clear name, email, and message on successful submit", () => {
    const props = {
      onSubmit: jest.fn(),
    };

    const wrapper = mountFormWithInputs(
      <ContactForm {...props} />,
      {
        contactName: "user12",
        contactEmail: "user12@test.com",
        contactMessage: "a message to send.",
      },
      testStore,
    );
    const contactNameNode = wrapper.find("#contactName");
    const contactNameErrorNode = wrapper.find("#contactNameError");

    const contactEmailNode = wrapper.find("#contactEmail");
    const contactEmailErrorNode = wrapper.find("#contactEmailError");

    const contactMessageNode = wrapper.find("#contactMessage");
    const contactMessageErrorNode = wrapper.find("#contactMessageError");

    const submitButton = wrapper.find("#contactSubmit");

    // give focus to contactMessageNode input (like a user would do before submit)
    contactEmailNode.prop("onFocus")();
    submitButton.simulate("click");

    //
    // NOTE:  Skipping this test because can't get the form field values
    //        to update correctly!!!
    //        It does work in the actual code. And it does work with older
    //        enzyme & react packages in edh-app
    //
    expect(contactNameNode.props().value).toEqual("");
    expect(contactNameErrorNode.length).toBe(0);

    expect(contactEmailNode.props().value).toEqual("");
    expect(contactEmailErrorNode.length).toBe(0);

    expect(contactMessageNode.props().value).toEqual("");
    expect(contactMessageErrorNode.length).toBe(0);

    // NOTE: document.querySelector called in onSubmitSuccess works in browser but not in jest
    // expect(usernameNode.get(0)).toEqual(document.activeElement);
  });
});
