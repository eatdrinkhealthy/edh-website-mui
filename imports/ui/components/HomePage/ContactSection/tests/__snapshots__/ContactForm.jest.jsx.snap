// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm /> matches render snapshot - with no form validation errors 1`] = `
<div>
  <h4
    className="MuiTypography-root-2 MuiTypography-h4-17 MuiTypography-colorPrimary-32"
  >
    Contact Us
  </h4>
  <form
    id="contactForm"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root-37 MuiFormControl-fullWidth-40"
      onDragStart={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "margin": "10px 0 10px 0",
        }
      }
    >
      <label
        className="MuiFormLabel-root-52 MuiInputLabel-root-41 MuiInputLabel-formControl-46 MuiInputLabel-animated-49 MuiInputLabel-outlined-51"
        data-shrink={false}
      >
        Name
      </label>
      <div
        className="MuiInputBase-root-72 MuiOutlinedInput-root-59 MuiInputBase-fullWidth-81 MuiInputBase-formControl-73"
        onClick={[Function]}
      >
        <fieldset
          aria-hidden={true}
          className="MuiPrivateNotchedOutline-root-89 MuiOutlinedInput-notchedOutline-66"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="MuiPrivateNotchedOutline-legend-90"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
        <input
          aria-invalid={false}
          className="MuiInputBase-input-82 MuiOutlinedInput-input-67"
          disabled={false}
          name="contactName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-37 MuiFormControl-fullWidth-40"
      onDragStart={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "margin": "10px 0 10px 0",
        }
      }
    >
      <label
        className="MuiFormLabel-root-52 MuiInputLabel-root-41 MuiInputLabel-formControl-46 MuiInputLabel-animated-49 MuiInputLabel-outlined-51"
        data-shrink={false}
      >
        Email
      </label>
      <div
        className="MuiInputBase-root-72 MuiOutlinedInput-root-59 MuiInputBase-fullWidth-81 MuiInputBase-formControl-73"
        onClick={[Function]}
      >
        <fieldset
          aria-hidden={true}
          className="MuiPrivateNotchedOutline-root-89 MuiOutlinedInput-notchedOutline-66"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="MuiPrivateNotchedOutline-legend-90"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
        <input
          aria-invalid={false}
          className="MuiInputBase-input-82 MuiOutlinedInput-input-67 MuiInputBase-inputType-85"
          disabled={false}
          name="contactEmail"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-37 MuiFormControl-fullWidth-40"
      onDragStart={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "margin": "10px 0 10px 0",
        }
      }
    >
      <label
        className="MuiFormLabel-root-52 MuiInputLabel-root-41 MuiInputLabel-formControl-46 MuiInputLabel-animated-49 MuiInputLabel-outlined-51"
        data-shrink={false}
      >
        Message
      </label>
      <div
        className="MuiInputBase-root-72 MuiOutlinedInput-root-59 MuiInputBase-fullWidth-81 MuiInputBase-formControl-73 MuiInputBase-multiline-80 MuiOutlinedInput-multiline-65"
        onClick={[Function]}
      >
        <fieldset
          aria-hidden={true}
          className="MuiPrivateNotchedOutline-root-89 MuiOutlinedInput-notchedOutline-66"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="MuiPrivateNotchedOutline-legend-90"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
        <textarea
          aria-invalid={false}
          className="MuiInputBase-input-82 MuiOutlinedInput-input-67 MuiInputBase-inputMultiline-84 MuiOutlinedInput-inputMultiline-69"
          disabled={false}
          name="contactMessage"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          rows={10}
          value=""
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root-117 MuiButton-root-91 MuiButton-fab-108 MuiButton-extendedFab-109 MuiButton-contained-102 MuiButton-containedPrimary-103 MuiButton-raised-105 MuiButton-raisedPrimary-106 MuiButton-sizeLarge-115 ContactFormComponent-formButton-1"
      disabled={false}
      id="contactSubmit"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-92"
      >
        Send
      </span>
      <span
        className="MuiTouchRipple-root-120"
      />
    </button>
  </form>
</div>
`;
